lib/drives/pages/drive_chat_page.dart

```diff
@@ -16,107 +17,118 @@ class DriveChatPage extends StatefulWidget {
+    List<int> ids = widget.drive.chats!.map((Chat chat) => chat.id!).toList();
+    _messagesStream =
+        SupabaseManager.supabaseClient.from('messages').stream(primaryKey: ['id']).order('created_at').map(
+              (messages) =>
+                  Message.fromJsonList(messages.where((element) => ids.contains(element['chat_id'])).toList()),
+            );
     super.initState();
-    setState(() {
-      _drive = widget.drive;
-    });
-    loadDrive();
+  }
+
+  @override
+  void dispose() {
+    _messagesStream.drain();
+    super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    List<Widget> widgets = [];
-    if (_fullyLoaded) {
-      Set<Profile> riders = _drive!.approvedRides!.map((ride) => ride.rider!).toSet();
-      List<Ride> pendingRides = _drive!.pendingRides!.toList();
-      if (riders.isEmpty && pendingRides.isEmpty) {
-        return Scaffold(
-          appBar: AppBar(
-            title: Text(S.of(context).pageDriveChatTitle),
-          ),
-          body: Center(
-            child: Text(S.of(context).pageDriveChatEmptyMessage),
-          ),
-        );
-      } else {
-        if (riders.isNotEmpty) {
-          List<Widget> riderColumn = [
-            const SizedBox(height: 5.0),
-            Text(
-              S.of(context).pageDriveChatRiderHeadline,
-              style: Theme.of(context).textTheme.titleLarge,
-            ),
-            const SizedBox(height: 10.0),
-            _riderList(riders),
-          ];
-          widgets.addAll(riderColumn);
-        }
-        widgets.add(const SizedBox(height: 10.0));
-      }
-    } else {
-      widgets.add(const SizedBox(height: 10));
-      widgets.add(const Center(child: CircularProgressIndicator()));
-    }
+    List<Chat> chats = widget.drive.chats!.toList();
     return Scaffold(
       appBar: AppBar(
         title: Text(S.of(context).pageDriveChatTitle),
       ),
-      body: RefreshIndicator(
-        onRefresh: loadDrive,
-        child: ListView.separated(
-          itemCount: widgets.length,
-          itemBuilder: (context, index) {
-            return widgets[index];
-          },
-          separatorBuilder: (BuildContext context, int index) {
-            return const SizedBox(height: 10);
-          },
-        ),
-      ),
+      body: chats.isNotEmpty
+          ? StreamBuilder<List<Message>>(
+              stream: _messagesStream,
+              builder: (BuildContext context, AsyncSnapshot<List<Message>> snapshot) {
+                if (snapshot.hasData) {
+                  for (Message message in snapshot.data!) {
+                    Chat chat = chats.firstWhere((Chat chat) => chat.id == message.chatId);
+                    if (chat.messages!.contains(message)) {
+                      chat.messages!.remove(message);
+                    }
+                    chat.messages!.add(message);
+                  }
+                  List<Card> widgets = chats.map((chat) => _buildChatWidget(chat)).toList();
```

Comment:
I can see the chats for pending rides here. But that should not be allowed, right? I thought we wanted to add the chat only when the ride is approved?

Reply:
And then clicking on the chat, it shows the normal "Nichts zu sehen" Page. I can type, but my messages are not shown.

Reply:
(I had assumed you had fixed that when you wrote that chat was finished). Or is it just not pushed?

Reply:
we dicussed that the chat should be created when the ride is requested if I'm correct. Thats what the database is doing right now. We can change that so the chat only gets created when the ride is approved (would be better in my opinion anyways). For now i can just remove that the chat is shown

Reply:
With regards to events, the chat should absolutely be created after the ride has been requested, but they should only be shown to riders at that stage I think

Reply:
yes i changed it to that. But still don't agree with you we can discuss it in the next meeting